junit-jupiter-api/src/main/java/org/junit/jupiter/api/parallel/Isolated.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api.parallel;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.apiguardian.api.API;
+
+@API(status = EXPERIMENTAL, since = "5.7")
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Inherited
+@ResourceLock("__global__")
```

Comment:
❓ Feedback from JUnit team call: maybe call the lock "isolated" as well?

Reply:
IMHO the name of the lock is fitting, as the lock is used by _every_ `TestDescriptor`, I would find it more confusing if every such descriptor would acquire a read lock on `__isolated__`. 
Just as a side note: https://en.wikipedia.org/wiki/Global_interpreter_lock

Maybe we could change the annotation to `@GlobalIsolation` if `@Isolated` to `__global__` is too surprising.

Reply:
I went with `org.junit.platform.engine.support.hierarchical.ExclusiveResource.GLOBAL_KEY` now for consistency with the constants in the `Resources` class.